src/app/features/asset-list/stacks/sip10-token-asset-list/sip10-token-asset-item.tsx

```diff
@@ -1,36 +1,49 @@
+import type { CryptoAssetBalance, MarketData, Sip10CryptoAssetInfo } from '@leather-wallet/models';
+
+import { convertAssetBalanceToFiat } from '@app/common/asset-utils';
+import { getSafeImageCanonicalUri } from '@app/common/stacks-utils';
+import { spamFilter } from '@app/common/utils/spam-filter';
 import { CryptoAssetItemLayout } from '@app/components/crypto-asset-item/crypto-asset-item.layout';
 import { StacksAssetAvatar } from '@app/components/stacks-asset-avatar';
-import type {
-  AccountCryptoAssetWithDetails,
-  Sip10AccountCryptoAssetWithDetails,
-} from '@app/query/models/crypto-asset.model';
-
-import { parseSip10TokenCryptoAssetBalance } from './sip10-token-asset-item.utils';
 
 interface Sip10TokenAssetItemProps {
-  asset: Sip10AccountCryptoAssetWithDetails;
-  onClick?(asset: AccountCryptoAssetWithDetails): void;
+  balance: CryptoAssetBalance;
+  info: Sip10CryptoAssetInfo;
+  isLoading: boolean;
+  marketData: MarketData;
+  onClick?(symbol: string, contractId?: string): void;
```

Comment:
`onSelectAsset` or similar descriptive name is better then repeating the default event names imo